backend/typescript/services/implementations/schedulingService.ts

```diff
@@ -69,12 +70,32 @@ class SchedulingService implements ISchedulingService {
 
   async getSchedulingsByDonorId(
     donorId: string,
+    weekLimit?: number,
   ): Promise<Array<SchedulingDTO>> {
     let schedulingDtos: Array<SchedulingDTO> = [];
+    let schedulings: Array<Scheduling>;
     try {
-      const schedulings: Array<Scheduling> = await Scheduling.findAll({
-        where: { donor_id: Number(donorId) },
-      });
+      if (weekLimit) {
+        const currentStartDate = new Date();
+        const nextDate = new Date();
+        nextDate.setDate(nextDate.getDate() + weekLimit * 7);
+        schedulings = await Scheduling.findAll({
+          where: {
+            donor_id: Number(donorId),
+            start_time: {
+              [Op.between]: [currentStartDate, nextDate] as any,
```

Comment:
```suggestion
              [Op.between]: [currentStartDate, nextDate],
```
don't think you need that